std/numeric.d

```diff
@@ -2928,19 +2928,20 @@ an efficient algorithm such as $(HTTPS en.wikipedia.org/wiki/Euclidean_algorithm
 or $(HTTPS en.wikipedia.org/wiki/Binary_GCD_algorithm, Stein's) algorithm.
 
 Params:
-    T = Any numerical type that supports the modulo operator `%`. If
-        bit-shifting `<<` and `>>` are also supported, Stein's algorithm will
+    a = Integer value of any numerical type that supports the modulo operator `%`.
+        If bit-shifting `<<` and `>>` are also supported, Stein's algorithm will
         be used; otherwise, Euclid's algorithm is used as _a fallback.
+    b = Integer value of any equivalent numerical type.
 
 Returns:
     The greatest common divisor of the given arguments.
  */
-T gcd(T)(T a, T b)
-if (isIntegral!T)
+typeof(Unqual!(T).init % Unqual!(U).init) gcd(T, U)(T a, U b)
+if (isIntegral!T && isIntegral!U)
 {
-    static if (is(T == const) || is(T == immutable))
+    static if (is(T == const) || is(T == immutable) || is(U == const) || is(T == immutable))
```

Comment:
```suggestion
    static if (is(T == const) || is(T == immutable) || is(U == const) || is(U == immutable))
```

Reply:
Ah yes, I was meant to test mixing const/immutable as well.

Reply:
There's no need to actually have this, using `CommonType(Unqual!T, Unqual!U)` instead for holding the intermediaries.